merkle/merkle_path.go

```diff
@@ -231,8 +354,38 @@ func skipMissingLevels(snapshot, lastNode int64, level int, node int64) (int, in
 		level--
 		sibling *= 2
 		lastNode = (snapshot - 1) >> uint(level)
-		glog.V(vLevel).Infof("Move down: S:%d L:%d LN:%d", sibling, level, lastNode)
+		glog.V(vvLevel).Infof("Move down: S:%d L:%d LN:%d", sibling, level, lastNode)
 	}
 
 	return level, sibling
 }
+
+// checkRecomputation carries out an additional check that the results of recomputePastSnapshot
+// are valid. There must be at least one fetch. All fetches must have the same rehash state and if
+// there is only one fetch then it must not be a rehash. If all checks pass then the fetches
+// represent one node after rehashing is completed.
+func checkRecomputation(fetches []NodeFetch) error {
```

Comment:
We've got a lot of defensive checks in the existing log implementation and we're likely to carry most of them over. The cost of serving bad data is potentially scrapping the log.